backend/src/main/java/com/elshop/backend/product/ProductController.java

```diff
@@ -0,0 +1,58 @@
+package com.elshop.backend.product;
+
+
+import com.elshop.backend.common.AppUtils;
+import com.elshop.backend.common.S3Service;
+import com.elshop.backend.product.model.Product;
+import com.elshop.backend.product.model.request.ProductRequest;
+import com.elshop.backend.product.model.response.ProductsListResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.validation.Valid;
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/api/products")
+@RequiredArgsConstructor
+public class ProductController {
+
+    private final ProductService productService;
+    private final S3Service s3Service;
+    private final AppUtils appUtils;
+
+    @GetMapping
+    public ResponseEntity<ProductsListResponse> getAllProducts(@RequestParam(defaultValue = "0") int page,
+                                                               @RequestParam(defaultValue = "10") int size) {
+        return new ResponseEntity<>(productService.getAll(page, size), HttpStatus.OK);
```

Comment:
Die ResponseEntity kann hier entfallen - dadurch wird der Code kürzer / leichter verständlich / vielleicht auch einfacher wiederzuverwenden.
```suggestion
    public ProductsListResponse getAllProducts(@RequestParam(defaultValue = "0") int page,
                                               @RequestParam(defaultValue = "10") int size) {
        return productService.getAll(page, size);
```

Hier nur einmalig, exemplarisch genannt - gilt aber auch für die weiteren Fälle im Code.